From 0024dcb08ea86ab83f1f83fccf19376d3b90cef3 Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Mon, 28 Aug 2017 17:23:15 +0200
Subject: [PATCH 1228/1345] arm64: dts: marvell: Force 3B addressing mode for
 SPI on A7k/A8k DB

commit  f0053a9b759549af3a7c4c58e136dde9f1fd348b from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Armada 7040/8040 DB boards can be shipped with various models of
boot SPI flash devices. An issue can emerge, when their size
exceeds 16MB - in such case the kernel driver will switch automatically
to 4B addressing mode. Later, in case of soft reset, the Boot ROM
will fail to fetch the firmware during the board initialization.

In order to prevent such behavior, this patch forces 3B addressing
mode used on affected boards by adding newly introduced
'spi-3byte-addressing' property to the SPI flash nodes.

Change-Id: I0043deac19c855553bd53506375cc184be8a0db0
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/43589
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-7040-db.dtsi |    1 +
 arch/arm64/boot/dts/marvell/armada-8040-db.dtsi |    1 +
 2 files changed, 2 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
index 73a3a17..5b001d7 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
@@ -117,6 +117,7 @@
 					compatible = "jedec,spi-nor";
 					reg = <0x0>;
 					spi-max-frequency = <20000000>;
+					spi-3byte-addressing;
 					partition@0 {
 						label = "boot";
 						reg = <0x0 0x200000>;
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
index f4b328a..c14687b 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
@@ -248,6 +248,7 @@
 					compatible = "jedec,spi-nor";
 					reg = <0x0>;
 					spi-max-frequency = <20000000>;
+					spi-3byte-addressing;
 					partition@0 {
 						label = "boot";
 						reg = <0x0 0x200000>;
-- 
1.7.9.5

